x86/64: Add Intel SOC Arizona Beach platform.

jialelux opened the pull request:
Arizona Beach is a derivative of the Alder Lake-M series product targeted specifically at networking applications.
Nexsec MT110 is a uCPE power by Intel Atom SOC Arizona Beach

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

hauke commented on target/linux/x86/image/64.mk:
```diff
@@ -7,4 +7,15 @@ define Device/generic
+  DEVICE_VENDOR := Intel Arizona Beach (NEXSEC MT110)
+  DEVICE_PACKAGES += \
+	kmod-e1000e kmod-e1000 kmod-fs-vfat kmod-igb kmod-ixgbe kmod-igc kmod-i40e \
+	kmod-iwlwifi kmod-usb-net-rndis kmod-usb-net-cdc-mbim kmod-i2c-i801
```
Does this device has a kmod-i40e (Intel 40G) network card?

jialelux replied:
MT110 is a Low Power,High Performance  network appliance. igc/igb in board , and i40e (10G/40G) for PCIe. We trying to run vpp/dpdk  on openwrt.

ynezz replied:
> kmod-i2c-i801 kmod-i40e kmod-iwlwifi kmod-usb-net-cdc-mbim kmod-usb-net-rndis

I'm fine with adding those packages to generic target.

jialelux replied:
Thanks! I will continue to improve.

hauke commented:
What is the advantage of this profile over the generic one?

jialelux commented:
> What is the advantage of this profile over the generic one?

first we will add the drivers of hardware on board. Second , We will optimize the hardware performance of AZB. We also try to add vpp/dpdk to openwrt. But this is not a simple issue.

intelerxia commented:
> What is the advantage of this profile over the generic one?
Hi hauke
Thanks for the reply.
Since the feature and device based on Intel's SOC platform grows faster, we think a generic config file maybe is not enough anymore.  We also check other architecture except x86 in OpenWRT target folder, it seems some architectures have supported many config files for different chips and different boards, but x86 only contains a generic config file, it is difficult for some customers not familiar with OpenWRT to config it, on the other hand, we have many customers who want to enable most part of the feature on specific SOC and board, we want to upload these config files which have been provided to part of the customers to OpenWRT community.   
Previously, we have submit a C3000 patch (pull 10026) and have some discussion there, you can take as a reference, since AZB is more latest than C3000 on intel's roadmap and the customers are grow faster, we submit the new patch here for this AZB.
Thanks
Yang

jialelux committed:
x86/64: Add Intel SOC Arizona Beach platform.

Arizona Beach is a derivative of the Alder Lake-M series product targeted specifically at networking applications.
Nexsec MT110 is a uCPE power by Intel Atom SOC Arizona Beach

Signed-off-by: Lu Jiale <jialex.lu@intel.com>

intelerxia commented:
> What is the advantage of this profile over the generic one?

Hi Hauk, is any more question? I have a question about how to add more reviewer in this pull request? Thanks

ynezz commented:
> first we will add the drivers of hardware on board. 

Sure, let's add those drivers for hardware shipped on board into generic profile.

> Second , We will optimize the hardware performance of AZB.

Please kindly note, that OpenWrt is not going to produce another x86/64 toolchain with board/family specific compiler options.

> we think a generic config file maybe is not enough anymore

It's still good enough for generic use cases. You need to provide us with patches, which would prove, that it can't be handled with generic profile anymore, then we should probably reconsider.

> We also check other architecture except x86 in OpenWRT target folder, it seems some architectures have supported many config files for different chips and different boards, 

Because it was proven, that OpenWrt wouldn't be able to support such boards, due to some constraints like for example smaller flash memory etc. With x86/64 we usually don't have such constraints, so we can support all such boards with single generic profile. It works quite well so far.

> Previously, we have submit a C3000 patch (pull 10026) and have some discussion there, you can take as a reference, since AZB is more latest than C3000 on intel's roadmap and the customers are grow faster, we submit the new patch here for this AZB.

IMO the discussion in #10026 makes it clear, that x86/64 generic target with additional kernel modules is a right way forward.

jialelux committed:
kernel: Add patch for Fibocom 5G module FM350

Add serial for MEDIATEK FM350-GL module support.
FM350 base on MEDIATEK T700 SOC.
usb-devices output for 0x7127.

Signed-off-by: Lu Jiale <jialex.lu@intel.com>

hauke commented on target/linux/x86/patches-5.15/120-usb-net-for-fm350.patch:
```diff
@@ -0,0 +1,59 @@
+---
+===================================================================
+--- linux-5.10.108.orig/drivers/usb/serial/option.c	2022-09-23 14:27:17.184479952 +0800
++++ linux-5.10.108/drivers/usb/serial/option.c	2022-09-23 14:33:22.907036605 +0800
```
Please run `make target/linux/{clean,refresh} V=99`  to refresh this patch

hauke commented on target/linux/x86/patches-5.15/120-usb-net-for-fm350.patch:
```diff
@@ -0,0 +1,59 @@
+ 
+ 
+ static const struct usb_device_id option_ids[] = {
++	{ USB_DEVICE(MEDIATEK_VENDOR_ID, 0x7127) },
```
If you only want to support the interface number 6 I would prefer something like this:
````
{ USB_DEVICE_INTERFACE_NUMBER(MEDIATEK_VENDOR_ID, 0x7127, 0x6) },
````
See here for the matcher: https://elixir.bootlin.com/linux/v5.15.80/source/include/linux/usb.h#L1015

hauke commented on target/linux/x86/patches-5.15/120-usb-net-for-fm350.patch:
```diff
@@ -0,0 +1,59 @@
+Author: Ji Li <liji@huachen.link>
+Date:   Thu Jun 23 16:56:44 2022 +0800
+
+    USB: serial: option: add MEDIATEK FM350-GL module support
```
Please send this patch also to upstream Linux. OpenWrt is fine to take it already before it is integrated in upstream, but we prefer patches to be inside of upstream Linux, then OpenWrt does not have to maintain them forever. 
There are some internal process at Intel to help with Linux upstreaming, this change looks pretty simple so I do not think it will be hard. There was an internal mailing list at Intel for such reviews, maybe Andy Shevchenko ( andriy.shevchenko@linux.intel.com ) can help you with that.

hauke commented:
This looks fine to me in general, I just had some comments on the patch.
